src/certificate.js

```diff
@@ -406,18 +429,25 @@ export default class Certificate {
       return;
     }
 
-    let readableAction = getVerboseMessage(step);
-    log(readableAction);
-    this._updateStatusCallback(step, readableAction, Status.starting);
+    let readableAction;
+    if (step) {
+      readableAction = SUB_STEPS.language[step].actionLabel;
+      log(readableAction);
+      this._updateStatusCallback(step, readableAction, VERIFICATION_STATUSES.STARTING);
+    }
 
     try {
       let res = await action();
-      this._updateStatusCallback(step, readableAction, Status.success);
-      this._stepsStatuses.push({step, status: Status.success, readableAction});
+      if (step) {
+        this._updateStatusCallback(step, readableAction, VERIFICATION_STATUSES.SUCCESS);
+        this._stepsStatuses.push({step, status: VERIFICATION_STATUSES.SUCCESS, readableAction});
+      }
       return res;
     } catch (err) {
-      this._updateStatusCallback(step, readableAction, Status.failure, err.message);
-      this._stepsStatuses.push({step, status: Status.failure, readableAction, message: err.message});
+      if (step) {
+        this._updateStatusCallback(step, readableAction, VERIFICATION_STATUSES.FAILURE, err.message);
+        this._stepsStatuses.push({step, status: VERIFICATION_STATUSES.FAILURE, readableAction, message: err.message});
+      }
```

Comment:
same here, but I guess you can even abstract the rest that does not go into the try to a function `postDoAction` and reduce code duplication.

Reply:
👍